### PR TITLE
Greeting and Thank-You Cards

### PR DESCRIPTION
**Pull request by SabreCat**
Implements two new cards people can send to their party members, similar to the New Year's and Valentine's cards, but available all year.
- Includes @crookedneighbor's refactor of greeting cards to display in a single modal with Markdown.
- Makes some existing card-related strings dynamic and translatable instead of hard-coded.
- Implements a notification when a user receives a card, which can be dismissed either by opening and accepting a card, or by dismissing it from the notifs dropdown.

### CODE REVIEW

File: website/public/js/controllers/menuCtrl.js [hunk 1]
```diff
@@ -41,7 +45,7 @@ angular.module('habitrpg')
       };
 
       $scope.hasNoNotifications = function() {
-        return selectNotificationValue(false, false, false, false, true);
+        return selectNotificationValue(false, false, false, false, false, true);
```
Comment (crookedneighbor) [inline website/public/js/controllers/menuCtrl.js:48]: uuuuuuuuuuuuuuuuuuuuuuuugh

Not something to do now, but this should be refactored to take in a hash instead.

File: test/common/user.ops.test.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      flags: { }
+    };
+
+    shared.wrap(user);
+  });
+
+  describe('readCard', function() {
+    it('removes card from invitation array', function() {
+      user.items.special.valentineReceived = ['Leslie'];
+      user.ops.readCard('valentine');
```
Comment (crookedneighbor) [inline test/common/user.ops.test.js]: Doesn't this need to be refactored to take a hash?

Comment (crookedneighbor) [inline test/common/user.ops.test.js]: s/refactored/adjusted

File: website/public/js/controllers/menuCtrl.js (hunk 1 above)
Comment (SabreCat) [inline website/public/js/controllers/menuCtrl.js:48]: Yeah, bit of a "lolwut" when I saw that. Not built in extensible fashion, then extended anyway.

File: test/common/user.ops.test.js (hunk 2 above)
Comment (SabreCat) [inline test/common/user.ops.test.js]: Prrrobably!

File: website/views/shared/profiles/achievements.jade [hunk 3]
```diff
@@ -226,6 +226,18 @@ div(ng-if='::profile.achievements.costumeContest')
     =env.t('costumeContestText')
   hr
 
+div(ng-if='::profile.achievements.greeting')
+  .achievement.achievement-greeting
+  h5=env.t('cheeryChum')
```
Comment (crookedneighbor) [inline website/views/shared/profiles/achievements.jade:231]: `cheeryChum` is cute, but we could refactor these into mixins if we renamed them something like `greetingCardAchievementTitle` and `greetingCardAchievementText`

Any reason this is a bad idea?

File: common/script/content.coffee [hunk 4]
```diff
@@ -951,10 +952,75 @@ api.spells =
...
+            t.achievements.thankyou++
+        if !target.items.special.thankyouReceived
+          target.items.special.thankyouReceived = []
+        target.items.special.thankyouReceived.push user.profile.name
+        target.flags.cardReceived = true
+
+        target.markModified? 'items.special.thankyouReceived'
+        user.stats.gp -= 10        
+        
+api.cardTypes =
```
Comment (crookedneighbor) [inline common/script/content.coffee:1008]: I have ideas for this. One second.

Comment (crookedneighbor) [inline common/script/content.coffee:1008]: Hmmm... my idea led me down a refactoring rabbit hole. Going to table it for now, but it's something I want to do in the near future.

The gist of it is to separate out the cards and special spells (spook dust, seafoam, etc) from the normal spells.

File: website/views/shared/profiles/achievements.jade (hunk 3 above)
Comment (SabreCat) [inline website/views/shared/profiles/achievements.jade:231]: Nope, sounds like a great idea to me!

File: common/script/content.coffee (hunk 4 above)
Comment (SabreCat) [inline common/script/content.coffee:1008]: Ah, put them into a different tree entirely from `spells`? That sounds like a good idea. We could get rid of a bunch of odd special handling. Though would we need to create a new API endpoint for them?

File: common/locales/en/limited.json [hunk 5]
```diff
@@ -33,16 +34,18 @@
...
     "nyeCardNotes": "Send a New Year's card to a party member.",
     "seasonalItems": "Seasonal Items",
     "auldAcquaintance": "Auld Acquaintance",
     "auldAcquaintanceText": "Happy New Year! Sent or received <%= cards %> New Year's cards.",
-    "newYear0": "Happy New Year! May you slay many a bad Habit.",
-    "newYear1": "Happy New Year! May you reap many Rewards.",
-    "newYear2": "Happy New Year! May you earn many a Perfect Day.",
-    "newYear3": "Happy New Year! May your To-Do list stay short and sweet.",
-    "newYear4": "Happy New Year! May you not get attacked by a raging Hippogriff.",
+    "nye0": "\"Happy New Year! May you slay many a bad Habit.\"",
```
Comment (crookedneighbor) [inline common/locales/en/limited.json:44]: Hmmmm.... we need to figure out why the escaped quotes are necessary. Something to do with the markdown, I expect.

**Pull request merged**